### ISSUE TITLE
discriminator loss

### ISSUE THREAD

**Issue by MengZhen-Chi** [SHI-Labs/StyleNAT#11]
in your paper you used no-saturated loss(soft-plus) for D and G, could you told me the change of D loss in your training process? In my training process, my D loss will down to 1.39 after about 1k iters

**Comment by stevenwalton** [comment]
I am a bit confused by your question. G uses a non-saturated loss (like stylegan) while the discriminator uses a logit loss and r1 regularization. 

As for your loss values, it is important to remember that for a GAN the generator and discriminator's losses are in competition with one another. You will not see a (mostly) smooth convergence to a minima like you would see in a classification model. The values will jump around  To help illustrate this, here's an example. Also, 1k iterations is not very many, and you're going to have a difficult time observing the minimax game between the networks. 

![image](https://github.com/SHI-Labs/StyleNAT/assets/4933582/683edac3-e5f1-4985-9d76-0cec140c4fcc)

**Comment by MengZhen-Chi** [comment]
I'm sorry that my description has left you puzzled. In my training process, the discriminator's loss rapidly drops to 1.39 within approximately 1000 iterations, while the generator's loss undergoes a gradual decline over an extended period to become comparable to the discriminator's.
because the logit loss = F.softplus(-real_pred) + F.softplus(fake_pred), 1.39 is equivalent to both real_pred and fake_pred being 0. 
In other words, in my training process, the discriminator assigns an output of 0 to both real and generated images.
My batch size is 4. I've looked up information, and it's possible that this could be a "mode collapse". Should I consider increasing the batch size?

**Comment by MengZhen-Chi** [comment]
And our tasks are also different. I'm using StyleNat to accomplish the restoration task, not a generation task. So, in the context of the restoration task, will the adversarial interplay between the discriminator and generator, as you mentioned, occur during GAN training?

**Comment by stevenwalton** [comment]
Discriminator loss being around 1.39 doesn't seem weird to me. You can see in the above image that this isn't abnormal.

Generator loss going to 0 is concerning. Your generator has collapsed. But I don't have enough information to help you out here. I'll need your config file and to understand the modifications you made to the network. If the problem is with my code I can debug but if the problem is in your code then that's your project.

GANs are notoriously unstable. I suggest __decreasing__ the batch size rather than increasing it. 

Can you tell me what changes you made? I need to understand what you did to provide help. It is not possible for me to provide advice simply from loss values. The more detail the better.

**Comment by stevenwalton** [comment]
Conversation moved to twitter. Closing. Please reopen if we move the conversation back here.

For posterity: G loss is close to D loss (not zero), significant changes to model, and straining from scratch
